.github/workflows/official-pr.yml

```diff
@@ -0,0 +1,63 @@
+name: Create official images PR
+
+on:
+  pull_request_target:
+    types:
+      - closed
```

Comment:
Is `pull_request_target` the best event for this? Why is it better than `push`?

Reply:
I originally had `push` and did an API request to fetch the associated PRs, but I switched to `pull_request_target` because the context included the PR number.

Reply:
I see. The downside is that any of the 15 other PRs open right now won't trigger this event unless they are rebased before merge. It also doesn't trigger if someone pushes directly to master.

I'm wondering if: a) we can replace `github.event.pull_request.title` with the commit message; and b) for pushes coming from PRs, if `github.event.pull_request` is available (would need to test it in a separate repo).

Not blocking and I'm not familiar with this repo commit workflow, so @nodejs/docker folks will know better which event to use. But `pull_request_target` has drawbacks, just pointing those so the team makes the right decision.

Reply:
If this is merged before the other PRs, I am thinking it will still be triggered because `master` will have this action and should detect a PR being closed. I am unsure of these semantics. I will run a quick test to see what happens.

Reply:
ah right, _maybe_ it works. `pull_request_target` triggers are kinda meh though, so I wouldn't be surprised if it didn't work.

Reply:
Alright ran the test and it does trigger as long as it's in `master`. However, I agree with you and would prefer it being on `push` to `master` instead too. I'd prefer it to be a `push` trigger too. I think I could replicate the behavior with a version shorter than my original one by adding a step like:
```
      - uses: actions/github-script@v3
        id: pr
        with:
          result-encoding: string
          script: |
            const result = await github.repos.listPullRequestsAssociatedWithCommit({
              owner: context.payload.repository.owner.name,
              repo: context.payload.repository.name,
              commit_sha: context.payload.head_commit.id
            })
            return result.data[0].number;
```

> The main branch is protected, so it can only be changed through PRs. That should simplify some stuff for you 😄

This also gives me confidence that I wouldn't have to worry about pushes to `master` happening through anything else other than from a PR.

Reply:
Thanks for testing it. Yes, I missed the master branch is protected here, so we should be good :)